File: Practice problem/Find prime number/Find prime number.py
# 연습문제 소수 찾기 Level 1

from math import sqrt, ceil
def solution(n):
    primelist = []
    answer = 0
    for candidate in range(2, n + 1):
        is_prime = True
        root = ceil(sqrt(candidate))
        for prime in primelist:
            if prime > root:
                break
            if candidate % prime == 0:
                is_prime = False
                break
        if is_prime:
            primelist.append(candidate)
            answer += 1
    return answer

File: Practice problem/Find prime number/Find prime number.java
# 연습문제 소수 찾기 Level 1

class Solution {
    public int solution(int n) {
        int[] primelist = new int[500000];
        int answer = 0;
        boolean is_prime = true;
        int root = 0;
        for(int i = 2; i < n+1 ; i++){
            is_prime = true;
            root = (int) Math.ceil(Math.sqrt(i));
            for(int j = 0; primelist[j] != 0 && primelist[j] <= root ; j++){
                if (i % primelist[j] == 0) {
                    is_prime = false;
                    break;
                }
            }
            if (is_prime) {
                primelist[answer] = i;
                answer++;
            }
        }
        return answer;
    }
}
